windows/src/engine/keyman32/kmprocess.cpp

```diff
@@ -141,7 +142,88 @@ BOOL ProcessHook()
 			_td->app->QueueDebugInformation(QID_BEGIN_ANSI, NULL, NULL, NULL, NULL, (DWORD_PTR) &keyinfo);
 	}
 
-	ProcessGroup(gp);
+  if (isUsingCoreProcessor) {
+    PWSTR buf;
+    buf = _td->app->ContextBufMax(MAXCONTEXT);
+    km_kbp_context_item *citems = nullptr;
+    ContextItemsFromAppContext(buf, &citems);
+    km_kbp_context_set(km_kbp_state_context(_td->state.lpActiveKBState), citems);
+    km_kbp_context_items_dispose(citems);
+    km_kbp_process_event(_td->state.lpActiveKBState, _td->state.vkey, static_cast<uint16_t>(Globals::get_ShiftState()));
+  }
+  else {
+    ProcessGroup(gp); // TODO: 5442 remove
+  }
+
+  
+  // Process the action items from the core. This actions will modify the windows context (AppContext).
+  // Therefore it is not required to copy the context from the core to the windows context.
+  for (auto act = km_kbp_state_action_items(_td->state.lpActiveKBState, nullptr); act->type != KM_KBP_IT_END; act++) {
+
+    switch (act->type)
+    {
+    case KM_KBP_IT_END:
+      assert(false);
+      break;
+      case KM_KBP_IT_ALERT:
+        _td->app->QueueAction(QIT_BELL, 0);
+        break;
+      case KM_KBP_IT_CHAR:
+       if (Uni_IsSMP(act->character)) {
+          _td->app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate1(act->character)));
+          _td->app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate2(act->character)));
+        }
+        else {
+          _td->app->QueueAction(QIT_CHAR, act->character);
+        }
+        break;
+      case KM_KBP_IT_MARKER:
+          _td->app->QueueAction(QIT_DEADKEY, (DWORD)act->marker);
+        break;
+      case KM_KBP_IT_BACK:
+        _td->app->QueueAction(QIT_BACK,BK_DEADKEY);
+        break;
+      case KM_KBP_IT_PERSIST_OPT:
+        if (act->option != NULL)
+        {
+          // Allocate for 1 option plus 1 pad struct of 0's for KM_KBP_IT_END
+          km_kbp_option_item* keyboardOpts = new km_kbp_option_item[2];
+          memmove(&(keyboardOpts[0]), act->option, sizeof(km_kbp_option_item));
+          km_kbp_status eventStatus = km_kbp_state_options_update(_td->state.lpActiveKBState, keyboardOpts);
+          if (eventStatus != KM_KBP_STATUS_OK)
+          {
+           // log warning "problem saving option for km_kbp_keyboard");
+            SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessHook: Problems saving optionf for keybooard [%s].", _td->lpActiveKeyboard->Name);
```

Comment:
```suggestion
            SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessHook: Error %d saving option for keyboard [%s].", eventStatus, _td->lpActiveKeyboard->Name);
```

We've got the return value, let's log it!

Reply:
This change has been made but now sits in `processcations` on line 61.